language/src/regent/codegen.t

```diff
@@ -2994,10 +3025,19 @@ local function make_partition_projection_functor(cx, expr, loop_index, color_spa
       [value.actions];
     end)
 
+    local parent = terralib.newsymbol(c.legion_logical_partition_t, "parent")
+    local index_access
+    if std.is_cross_product(std.as_read(ast.get_base_indexed_node(expr).expr_type)) then
```

Comment:
Don't we want this check to go the other way? If it's a partition then we wrap it. Cross products can fall under the "everything else" case.